contracts/metatx/MinimalForwarder.sol

```diff
@@ -23,59 +28,85 @@ contract MinimalForwarder is EIP712 {
         uint256 value;
         uint256 gas;
         uint256 nonce;
+        uint48 deadline;
         bytes data;
     }
 
-    bytes32 private constant _TYPEHASH =
-        keccak256("ForwardRequest(address from,address to,uint256 value,uint256 gas,uint256 nonce,bytes data)");
-
-    mapping(address => uint256) private _nonces;
+    bytes32 private constant _FORWARD_REQUEST_TYPEHASH =
+        keccak256(
+            "ForwardRequest(address from,address to,uint256 value,uint256 gas,uint256 nonce,uint48 deadline,bytes data)"
+        );
 
     /**
      * @dev The request `from` doesn't match with the recovered `signer`.
      */
     error MinimalForwarderInvalidSigner(address signer, address from);
 
     /**
-     * @dev The request nonce doesn't match with the `current` nonce for the request signer.
+     * @dev The request `value` doesn't match with the `msg.value`.
+     */
+    error MinimalForwaderMismatchedValue(uint256 msgValue, uint256 value);
+
+    /**
+     * @dev The list of requests length doesn't match with the list of signatures length.
      */
-    error MinimalForwarderInvalidNonce(address signer, uint256 current);
+    error MinimalForwaderInvalidBatchLength(uint256 requestsLength, uint256 signaturesLength);
 
-    constructor() EIP712("MinimalForwarder", "0.0.1") {}
+    /**
+     * @dev The request `deadline` has expired.
+     */
+    error MinimalForwarderExpiredRequest(uint256 deadline);
 
-    function getNonce(address from) public view returns (uint256) {
-        return _nonces[from];
-    }
+    /**
+     * @dev See {EIP7712-constructor}.
+     */
+    constructor(string memory name, string memory version) EIP712(name, version) {}
```

Comment:
In the documentation we should explain what the parameters are so the user can understand how to choose them.

I can see value in customizing the name, but I don't know why a user would want control over the version parameter for the contract, and I think it makes more sense for us to control it. We can hardcode it to an initial version `"1"` now, and if in a future version we want to change the behavior of the contract in some incompatible way, bumping this version to the next number could be a nice way to do that.

Reply:
> In the documentation we should explain what the parameters are so the user can understand how to choose them.

Honestly, I don't see how this adds value when the documentation will add a link to the explanation in EIP-712. Considering [this comment's context](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4346#discussion_r1235518479), do you still think an explanation is needed here?

> I can see value in customizing the name, but I don't know why a user would want control over the version parameter for the contract, and I think it makes more sense for us to control it.

Sure thing.